packages/core-components/src/layout/Sidebar/SidebarSubmenuItem.tsx

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useContext, useState } from 'react';
+import {
+  NavLink,
+  resolvePath,
+  useLocation,
+  useResolvedPath,
+} from 'react-router-dom';
+import { makeStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import Link from '@material-ui/core/Link';
+import { IconComponent } from '@backstage/core-plugin-api';
+import clsx from 'clsx';
+import { BackstageTheme } from '@backstage/theme';
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
+import ArrowDropUpIcon from '@material-ui/icons/ArrowDropUp';
+import { SidebarItemWithSubmenuContext } from './config';
+
+const useStyles = makeStyles<BackstageTheme>(theme => ({
+  item: {
+    height: 48,
+    width: '100%',
+    '&:hover': {
+      background: '#6f6f6f',
+      color: theme.palette.navigation.selectedColor,
+    },
+    display: 'flex',
+    alignItems: 'center',
+    color: theme.palette.navigation.color,
+    padding: 20,
+    cursor: 'pointer',
+    position: 'relative',
+    background: 'none',
+    border: 'none',
+  },
+  itemContainer: {
+    width: '100%',
+  },
+  selected: {
+    background: '#6f6f6f',
+    color: '#FFF',
+  },
+  label: {
+    margin: 14,
+    marginLeft: 7,
+    fontSize: 14,
+  },
+  dropdownArrow: {
+    position: 'absolute',
+    right: 21,
+  },
+  dropdown: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'end',
+  },
+  dropdownItem: {
+    width: '100%',
+    padding: '10px 0 10px 0',
+  },
+  textContent: {
+    color: theme.palette.navigation.color,
+    display: 'flex',
+    justifyContent: 'center',
+    fontSize: '14px',
+  },
+}));
+
+/**
+ * Clickable item displayed when submenu item is clicked.
+ * title: Text content of item
+ * to: Path to navigate to when item is clicked
+ *
+ * @public
+ */
+export type SidebarSubmenuItemDropDownItem = {
```

Comment:
```suggestion
export type SidebarSubmenuItemDropdownItem = {
```

Or we make it `dropDownItems`, but I think this makes more sense :p